### PR TITLE
Regenerate all languages after EMF bump

### PR DESCRIPTION
**Pull request by szarnekow**
Added scripts to relengutils that allow to generate multiple languages in one step

### CODE REVIEW

**Comment by szarnekow** [comment]
MWE2 files for test languages are still pending.
Yet I'm curious if the new mwe2 files work for you, too @cdietrich @LorenzoBettini  - mind the hint 
```
/**
 * Must run with -Dslf4j.provider=org.slf4j.simple.SimpleServiceProvider
 */
```

File: org.eclipse.xtend.core/src-gen/org/eclipse/xtend/core/parser/antlr/internal/InternalXtendFlexer.java [hunk 1]
```diff
@@ -892,7 +892,7 @@ public String getTokenText() {
     char [] map = new char[0x10000];
     int i = 0;  /* index in packed string  */
     int j = 0;  /* index in unpacked array */
-    while (i < 2336) {
```
Comment (szarnekow) [inline org.eclipse.xtend.core/src-gen/org/eclipse/xtend/core/parser/antlr/internal/InternalXtendFlexer.java:895]: Due to different unicode tables in Java 11 and Java 17 I suppose

File: org.eclipse.xtext.relengutil/META-INF/MANIFEST.MF [hunk 2]
```diff
@@ -6,7 +6,18 @@ Bundle-Version: 2.38.0.qualifier
 Bundle-Vendor: Eclipse Xtext
 Bundle-RequiredExecutionEnvironment: JavaSE-17
 Require-Bundle: org.eclipse.xtext.util;bundle-version="2.38.0",
- org.eclipse.m2e.maven.runtime;bundle-version="3.9.600"
+ org.eclipse.m2e.maven.runtime;bundle-version="3.9.600",
```
Comment (szarnekow) [inline org.eclipse.xtext.relengutil/META-INF/MANIFEST.MF:9]: m2e.maven.runtime registers a SLF4J thingy by means of a service loader. Thus we need the command line `-D` when running the mwe2 workflows

File: org.eclipse.xtext.relengutil/src/org/eclipse/xtext/relengutil/GenerateAllExamples.mwe2 [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+import org.eclipse.xtext.xtext.generator.*
+import org.eclipse.xtext.xtext.generator.model.project.*
+
+var rootPath = "../org.eclipse.xtext.xtext.ui.examples/projects"
+
+/**
+ * Must run with -Dslf4j.provider=org.slf4j.simple.SimpleServiceProvider
+ */
+Workflow {
```
Comment (szarnekow) [inline org.eclipse.xtext.relengutil/src/org/eclipse/xtext/relengutil/GenerateAllExamples.mwe2:15]: @cdietrich @LorenzoBettini Please let me know if this approach works for you

Comment (cdietrich) [reply org.eclipse.xtext.relengutil/src/org/eclipse/xtext/relengutil/GenerateAllExamples.mwe2:15]: will check in the evening

Comment (cdietrich) [reply org.eclipse.xtext.relengutil/src/org/eclipse/xtext/relengutil/GenerateAllExamples.mwe2:15]: wf works with that

Comment (cdietrich) [reply org.eclipse.xtext.relengutil/src/org/eclipse/xtext/relengutil/GenerateAllExamples.mwe2:15]: did not check diffs yet

Comment (szarnekow) [reply org.eclipse.xtext.relengutil/src/org/eclipse/xtext/relengutil/GenerateAllExamples.mwe2:15]: hmm there shouldn't be any. Curious.

Comment (szarnekow) [reply org.eclipse.xtext.relengutil/src/org/eclipse/xtext/relengutil/GenerateAllExamples.mwe2:15]: well .. I'm running with 2024-03 as TP

Comment (cdietrich) [reply org.eclipse.xtext.relengutil/src/org/eclipse/xtext/relengutil/GenerateAllExamples.mwe2:15]: same here.
there is also no diff towards your branch.

Comment (cdietrich) [reply org.eclipse.xtext.relengutil/src/org/eclipse/xtext/relengutil/GenerateAllExamples.mwe2:15]: => if you have committed all changes we are good

**Comment by LorenzoBettini** [comment]
@szarnekow just a quick note:

```
Error:  Cannot resolve dependencies of project org.eclipse.xtext:org.eclipse.xtext.relengutil:eclipse-plugin:2.38.0-SNAPSHOT
Error:   with context {osgi.os=macosx, osgi.ws=cocoa, org.eclipse.update.install.features=true, osgi.arch=x86_64, org.eclipse.update.install.sources=true}
Error:    Software being installed: org.eclipse.xtext.relengutil 2.38.0.qualifier
Error:    Missing requirement: org.eclipse.xtext.relengutil 2.38.0.qualifier requires 'osgi.bundle; org.eclipse.xtext.example.arithmetics 2.38.0' but it could not be found:
```

**Comment by LorenzoBettini** [comment]
Unfortunately, those examples are NOT Maven projects

**Comment by cdietrich** [comment]
maybe jsut make them optional deps?

File: org.eclipse.xtext.relengutil/pom.xml [hunk 4]
```diff
@@ -24,4 +26,24 @@
 			<version>${project.version}</version>
 		</dependency>
 	</dependencies>
+
+	<build>
+		<plugins>
+			<plugin>
+				<groupId>org.eclipse.tycho</groupId>
+				<artifactId>tycho-maven-plugin</artifactId>
```
Comment (LorenzoBettini) [inline org.eclipse.xtext.relengutil/pom.xml:34]: This plugin is useless: it's already enabled by the parent POM

Comment (szarnekow) [reply org.eclipse.xtext.relengutil/pom.xml:34]: Copy and paste mistake

**Review by cdietrich — APPROVED** [review]

**Comment by LorenzoBettini** [comment]
@cdietrich @szarnekow I don't know if that's related to this logging nightmare, but if you generate an xtext project with Maven support and run the Maven build, when running mwe2 you get a warning about missing log configuration and there's no logging output, though the run succeeds.

**Comment by cdietrich** [comment]
you mean pure maven or tycho?

**Comment by szarnekow** [comment]
I'll already merge this and take care of the testlanguages in the next step. Objections?

**Comment by LorenzoBettini** [comment]
> I'll already merge this and take care of the testlanguages in the next step. Objections?

Please go right ahead!

**Comment by LorenzoBettini** [comment]
> you mean pure maven or tycho?
> 
> ```
> [INFO] �[1m--- �[0;32mexec:3.0.0:java�[m �[1m(mwe2Launcher)�[m @ �[36morg.xtext.example.mydsl1�[0;1m ---�[m
> SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
> SLF4J: Defaulting to no-operation (NOP) logger implementation
> SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
> 0    [org.eclipse.emf.mwe2.launch.runtime.Mwe2Launcher.main()] INFO  text.xtext.generator.XtextGenerator  - Initializing Xtext generator
> 220  [org.eclipse.emf.mwe2.launch.runtime.Mwe2Launcher.main()] INFO  text.xtext.generator.XtextGenerator  - Generating org.xtext.example.mydsl1.MyDsl
> 260  [org.eclipse.emf.mwe2.launch.runtime.Mwe2Launcher.main()] INFO  nerator.ecore.EMFGeneratorFragment2  - Generating EMF model code
> 716  [org.eclipse.emf.mwe2.launch.runtime.Mwe2Launcher.main()] INFO  text.xtext.generator.XtextGenerator  - Generating common infrastructure
> 717  [org.eclipse.emf.mwe2.launch.runtime.Mwe2Launcher.main()] INFO  .emf.mwe2.runtime.workflow.Workflow  - Done.
> [INFO] 
> ```

Maven Tycho. Exactly the same warning.

**Pull request merged**